module/zfs/dsl_dir.c

```diff
@@ -1311,7 +1320,9 @@ dsl_dir_tempreserve_impl(dsl_dir_t *dd, uint64_t asize, boolean_t netfree,
 	 * If this transaction will result in a net free of space,
 	 * we want to let it through.
 	 */
-	if (ignorequota || netfree || dsl_dir_phys(dd)->dd_quota == 0)
+	if (ignorequota || netfree || dsl_dir_phys(dd)->dd_quota == 0 ||
+	    (dmu_objset_type(tx->tx_objset) == DMU_OST_ZVOL &&
+	    zfs_enforce_zvol_quotas == B_TRUE))
```

Comment:
Am I misreading this?  Shouldn't this be `zfs_enforce_zvol_quotas == B_FALSE` so we don't enforce the quota here?

Reply:
You are right.
Fixed.